Initial Checkin - Content Pack - Commvault Security IQ

Cv-securityIQ opened the pull request:
<!-- REMINDER: THIS IS A PUBLIC REPO DO NOT POST HERE SECRETS/SENSITIVE DATA -->

## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/demisto/content?pullRequest=27236) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/demisto/content?pullRequest=27236) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/demisto/content?pullRequest=27236) it.</sub>

Cv-securityIQ commented:
@RotemAmit with whom should I share the video for the Contribution Registration? I would need an email ID to provide access to the video.

RotemAmit commented:
> @RotemAmit with whom should I share the video for the Contribution Registration? I would need an email ID to provide access to the video.

Hi @Cv-securityIQ, you can share the video with me. I am available in DFIR as well.
Please fill the [CLA](https://cla-assistant.io/demisto/content?pullRequest=27236) form for @amitshakya-cv as well.

efelmandar reviewed (commented):
@Cv-securityIQ Thank you very much for your contribution, I finished reviewing the PR and overall it looks excellent, but a couple of points I do want to bring up are:

1. For all content items (incident types, fields, mapper, playbooks, and layout) please make sure to prefix the name with the pack name.

2. In the playbook, the `Disabling data aging` task returns a status response for the action, why not reflect it to the user or at least in the closing notes?

Please let me know if you have any further questions or need any assistance,

Cv-securityIQ commented:
We are currently making changes based on code review comments.

Cv-securityIQ commented:
> @Cv-securityIQ Thank you very much for your contribution, I finished reviewing the PR and overall it looks excellent, but a couple of points I do want to bring up are:
> 
> 1. For all content items (incident types, fields, mapper, playbooks, and layout) please make sure to prefix the name with the pack name.
> 2. In the playbook, the `Disabling data aging` task returns a status response for the action, why not reflect it to the user or at least in the closing notes?
> 
> Please let me know if you have any further questions or need any assistance,

We have another product called Metallic, for which we are intending to use the same integration. If we add Commvault Security in the prefixes it might cause some confusion later on.

maimorag commented:
> ecause we would prefer it if the users used tokens, instead of username password.

Please notice that "type:9" also can be utilized as a single parameter:
You can use a single parameter, as demonstrated in the following YAML example. In this case, the username is hidden, and the "Token" can be obtained by accessing -
```token = demisto.params().get("credentials", {}).get("password")```

- displaypassword: Token
  name:  credentials
  required: true
  hiddenusername: true
  type: 9

For more information regarding types you can check the integration "ipinfo_v2" line 9.

Cv-securityIQ commented:
We've changed the parameter type for the tokens (CommvaultAPIToken,AzureKeyVaultClientSecret) to type 9, and also added the company name as a prefix to all the custom fields, layout, incident type, playbook and mapper. Can you please review it  ?

edik24 commented:
Hi @Cv-securityIQ,
please see [here](https://xsoar.pan.dev/docs/concepts/design-best-practices) information about command names.

It should be:
PRODUCTNAME-OBJECTNAME-ACTION

Cv-securityIQ commented:
> Hi @Cv-securityIQ, please see [here](https://xsoar.pan.dev/docs/concepts/design-best-practices) information about command names.
> 
> It should be: PRODUCTNAME-OBJECTNAME-ACTION

Hi @edik24 , noted. We've changed the command names.

Cv-securityIQ commented:
@edik24 All the Review comments are handled and the pull request is updated.

efelmandar commented:
Hi @Cv-securityIQ I see you made the changes we discussed and it looks good, but I see you created an incident field named `Event ID` which already exists in the CommonTypes pack, please use the existing one as we try to minimize duplication of objects in the system

Cv-securityIQ commented:
Hi @efelmandar , @maimorag made the changes you suggested.

maimorag commented:
Few notes:
Unit tests must be present for all integrations/scripts commands (including fetch_incidents) and helper functions.
Missing unit tests( besides clients’ functions):
set_secret_in_key_vault
long-running-execution
fetch-incidents
field_mapper
2. please update the README - please run the `demisto-sdk generate-docs  -c  commvault-security-get-access-token-from-keyvault ` command (also, use the -e flag for using a commad_examples and add example for commvault-security-get-access-token-from coammand ) - for more information please see: https://xsoar.pan.dev/docs/concepts/demisto-sdk#generate-docs

Cv-securityIQ commented:
@maimorag , can you please review the changes ?

maimorag commented:
Please revert the changes in file .circleci/config.yml

Cv-securityIQ commented:
> Please revert the changes in file .circleci/config.yml

Do we need to make a change over here? Or Palo Alto team is making these changes internally?

maimorag commented:
Hi @Cv-securityIQ, the code looks good!

We're ready for a demo. Please check [this page](https://xsoar.pan.dev/docs/contributing/demo-prep), and let me and @edik24 know when you're available for one over DFIR.

maimorag commented on Packs/CommvaultSecurityIQ/Integrations/CommvaultSecurityIQ/CommvaultSecurityIQ.py:
```diff
@@ -0,0 +1,1473 @@
+            return_results("ok")
+        elif command == "fetch-incidents":
+            last_fetch, out = fetch_incidents(
+                client, last_run=demisto.getLastRun(), first_fetch_time=first_fetch_time
```
```suggestion
                client, last_run=demisto.getLastRun().get('last_fetch'), first_fetch_time=first_fetch_time
```

maimorag commented:
@Cv-securityIQ , We noticed the forked repo is named **Cv-securityIQ_master**  instead of the suggested 'content' naming convention.
This current situation is preventing us to continue with the merging process.
Could you please manually rename the repo using this manual?
https://docs.github.com/en/repositories/creating-and-managing-repositories/renaming-a-repository
Please notify me if there are any questions.
Thanks!

NEEHVN007 reviewed (commented):
Disable pipelines
